GitUI/CommandsDialogs/RevisionDiffControl.cs

```diff
@@ -1458,6 +1466,38 @@ private void ResetSelectedItemsWithConfirmation(bool resetToParent)
 
         private static bool RenamedIndexItem(FileStatusItem item) => item.Item.IsRenamed && item.Item.Staged == StagedStatus.Index;
 
+        private class ScriptOptionsProvider : IScriptOptionsProvider
+        {
+            private RevisionDiffControl _revisionDiffControl;
+
+            public ScriptOptionsProvider(RevisionDiffControl revisionDiffControl)
+            {
+                _revisionDiffControl = revisionDiffControl;
+            }
+
+            IReadOnlyList<string> IScriptOptionsProvider.Options => new[] { "SelectedFiles", "LineNumber" };
+
+            string IScriptOptionsProvider.GetValue(string option)
+            {
+                switch (option)
+                {
+                    case "SelectedFiles":
+                        IEnumerable<string>? selectedFiles = _revisionDiffControl.DiffFiles.SelectedItems.Select(item => _revisionDiffControl._fullPathResolver.Resolve(item.Item.Name));
+                        if (selectedFiles is null || !selectedFiles.Any())
+                        {
+                            return null;
+                        }
+
+                        return string.Join(" ", selectedFiles.Select(file => file.QuoteForCommandLine()));
+
+                    case "LineNumber":
+                        return $"{(_revisionDiffControl.BlameControl.Visible ? _revisionDiffControl.BlameControl.CurrentFileLine : _revisionDiffControl.DiffText.CurrentFileLine)}";
+                }
+
+                return null;
```

Comment:
The interface definition is not nullable. So either the implementation has to match the contract, or the contract needs to be made nullable. The latter is probably the better option.

Reply:
Accepted (Made the interface nullable, as it should have been in the first place)